lib/modules/versioning/nuget/version.ts

```diff
@@ -0,0 +1,68 @@
+import { regEx } from '../../../util/regex';
+import type { NugetVersion } from './types';
+
+// We can't just use `Number.parseInt()` beacause it parses `123abc` as `123`,
+// which leads to incorrect version comparison.
+function ensureNumber(input: string): number | null {
+  if (!regEx(/^\d+$/).test(input)) {
+    return null;
+  }
+
+  return Number.parseInt(input, 10);
+}
+
+function comparePrereleases(x: string, y: string): number {
+  const xParts = x.split('.');
+  const yParts = y.split('.');
+
+  const maxLen = Math.max(xParts.length, yParts.length);
+  for (let i = 0; i < maxLen; i += 1) {
+    const xPart = xParts[i] ?? '';
+    const yPart = yParts[i] ?? '';
+    const xNum = ensureNumber(xPart);
+    const yNum = ensureNumber(yPart);
+
+    const res =
+      xNum !== null && yNum !== null
+        ? Math.sign(xNum - yNum)
+        : xPart.localeCompare(yPart, undefined, { sensitivity: 'base' });
+
+    if (res !== 0) {
+      return res;
+    }
+  }
+
+  return 0;
+}
+
+export function compare(x: NugetVersion, y: NugetVersion): number {
+  const xMajor = x.major;
+  const yMajor = y.major;
+
+  const xMinor = x.minor ?? 0;
+  const yMinor = y.minor ?? 0;
+
+  const xPatch = x.patch ?? 0;
+  const yPatch = y.patch ?? 0;
+
+  const xRevision = x.revision ?? 0;
+  const yRevision = y.revision ?? 0;
+
+  if (xMajor !== yMajor) {
+    return Math.sign(xMajor - yMajor);
+  } else if (xMinor !== yMinor) {
+    return Math.sign(xMinor - yMinor);
+  } else if (xPatch !== yPatch) {
+    return Math.sign(xPatch - yPatch);
+  } else if (xRevision !== yRevision) {
+    return Math.sign(xRevision - yRevision);
+  } else if (x.prerelease && !y.prerelease) {
+    return -1;
+  } else if (!x.prerelease && y.prerelease) {
+    return 1;
+  } else if (x.prerelease && y.prerelease) {
+    return comparePrereleases(x.prerelease, y.prerelease);
```

Comment:
```suggestion
    return x.prerelease.localeCompare(y.prerelease, undefined, { sensitivity: 'base', numeric: true }));
```

why not simply using numeric[^1] compare?

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#numeric_sorting

Reply:
Short answer is "no, trust me". For longer answer, later I'll tell you which test will break.

Reply:
It breaks this test: https://github.com/renovatebot/renovate/blob/6dac20ec88f89e98b014b1263ff3405b8a5f3e89/lib/modules/versioning/nuget/index.spec.ts#L312

https://nugettools.azurewebsites.net/6.11.0-preview.2/version-comparison?versionA=1.2.0-beta2&versionB=1.2.0-beta10